go-controller/pkg/ovn/service.go

```diff
@@ -297,43 +237,32 @@ func (ovn *Controller) deleteService(service *kapi.Service) {
 	if !util.IsClusterIPSet(service) || len(service.Spec.Ports) == 0 {
 		return
 	}
-
-	ips := make([]string, 0)
-
 	for _, svcPort := range service.Spec.Ports {
 		var port int32
 		if util.ServiceTypeHasNodePort(service) {
 			port = svcPort.NodePort
 		} else {
 			port = svcPort.Port
 		}
-		if port == 0 {
-			continue
-		}
 
-		protocol, err := util.ValidateProtocol(svcPort.Protocol)
-		if err != nil {
+		if err := util.ValidatePort(svcPort.Protocol, port); err != nil {
 			klog.Errorf("Skipping delete for service port %s: %v", svcPort.Name, err)
 			continue
 		}
 
-		// targetPort can be anything, the deletion logic does not use it
-		var targetPort int32
 		if util.ServiceTypeHasNodePort(service) {
-			// Delete the 'NodePort' service from a load-balancer instantiated in gateways.
-			ovn.deleteGatewayVIPs(protocol, port)
+			ovn.deleteGatewayNodePortVIPs(svcPort.Protocol, port)
 		}
-		if util.ServiceTypeHasClusterIP(service) {
```

Comment:
why are you removing this check?

Reply:
Will rectify together with your previous comments in `syncServices`

Reply:
Actually no, I just realized it: the check `util.IsClusterIPSet(service)` does this for us already on the first line in the method.